File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
# 투포인터 사용
import sys
input = sys.stdin.readline

n = int(input())
arr = list(map(int, input().split()))
x = int(input())

arr.sort()

left, right = 0, n - 1

cnt = 0
while left < right:
    s = arr[left] + arr[right]
    if s == x : cnt += 1
    # 큰 경우 right - 1
    if s > x:
        right -= 1
    # 작거나 같은 경우
    else:
        left += 1
print(cnt)
    

File: 백준/Silver/3273. 두 수의 합/두 수의 합.js
let fs = require('fs')
let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n')
// let input = fs.readFileSync('example.txt').toString().trim().split('\n')

const maxNum = Number(input[2])
const arr = input[1]
  .split(' ')
  .map(Number)
  .sort((a, b) => a - b)

let s = 0
let e = Number(input[0]) - 1
let cnt = 0

while (s < e) {
  let num = arr[s] + arr[e]
  if (num == maxNum) {
    cnt += 1
    s += 1
  } else if (num < maxNum) {
    s += 1
  } else {
    e -= 1
  }
}

console.log(cnt)
